### ISSUE TITLE
ctas column definition violation not reported for snowflake

### ISSUE THREAD

**Issue by prabodh1194** [sqlfluff/sqlfluff#4016]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I have the following CTAS query for snowflake were the select clause defines 3 columns while the CTAS is expecting 4 columns.

```sql
create or replace temporary table aaa_magnetic_bag.public.zz (
    a int,
    b int,
    c int,
    d int
)
as
select
    a,
    b,
    c
from
    values (1, 2, 3) as x(a, b, c);
```

running lint on this query passes without any violations.

### Expected Behaviour

i was expecting an error regarding the column definition

### Observed Behaviour

running lint on this query passes without any violations.

### How to reproduce

```bash
cat << EOF > /tmp/c.sql
create or replace temporary table aaa_magnetic_bag.public.zz (
    a int,
    b int,
    c int,
    d int
)
as
select
    a,
    b,
    c
from
    values (1, 2, 3) as x(a, b, c);
EOF

sqlfluff lint --dialect snowflake /tmp/c.sql
```

### Dialect

snowflake

### Version

1.3.2

### Configuration

none

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
